apps/src/netsim/NetSimLogPanel.js

```diff
@@ -188,8 +188,19 @@ NetSimLogPanel.prototype.onClearButtonPress_ = function () {
 
 /**
  * Put a message into the log.
+ * @param {string} packetBinary
+ * @param {number} packetID
  */
-NetSimLogPanel.prototype.log = function (packetBinary) {
+NetSimLogPanel.prototype.log = function (packetBinary, packetID) {
+
+  var packetAlreadyInLog = this.packets_.some(function (packet) {
+    return packet.packetID_ == packetID;
```

Comment:
Although `NetSimLogPacket` is internal to this file, I'm not sure `NetSimLogPanel` ought to be accessing its private member `packetID_`.  For now, I'd suggest just making it public.